### PR TITLE
Add support to use GLFW (linux only for now);

### PR DESCRIPTION
**Pull request by fungos**
This implements a very thin and (ugly?)hacky layer that makes glfw behave as SDL, so to avoid a major rewrite of RecastDemo main.

I've added a premake option "--use-glfw" with a configuration for linux only (for now), adding support to windows and macosx should be trivial and need only changes to the premake file. I'm not doing this right now due to not being able to test it on these platforms.

Also note that the premake file changes may not be optimal, I still do not have a good premake knowledge.

Why GLFW?

Firstly because I've [implemented](https://github.com/fungos/fips-recast) support to [fips build system](https://github.com/floooh/fips) and at this moment we haven't a fipsified SDL build, only GLFW. 
But also, GLFW is an easier, good and lightweight alternative to SDL.

Why FIPS?

Fips is an amazing build system based on CMake where it manage all your (fips) dependencies automaticaly, having more and more fipsified projects means lower friction to programmers when adding and maintaining thirdparty dependencies. For example, commonly used "contrib" directories with instructions, git submodules, or just duplicated dependencies aren't needed anymore. Also, users using projects with this kind of dependencies will have not to worry anymore about Doing The Right Thing(tm).

### CODE REVIEW

**Comment by Kromster80** [comment]
I'm not sure I understand the logic behind it. Why there is a need replace SDL solution with GLFW in the first place?

**Comment by fungos** [comment]
This does not replace. GLFW is an alternative and optional.
The reasoning is described in the PR, though it may not be enough for some people.

TL;DR: 1) for use with a better build system than premake4 and; 2) GLFW for this kind of demo is better than SDL.

**Comment by memononen** [comment]
I like GLFW, I've used in most of my other projects. The right solution would be to porte the demo to use GLFW. GLFW also pretty much requires GLEW in windows, you can take a look at nanovg on how to set that up properly [1]. Supporting the library takes considerable amount of my free time and I dont want to maintain a hack.

[1] https://github.com/memononen/nanovg

**Comment by fungos** [comment]
Nice to hear you're open to GLFW. So I will do a clean port as I too do think is the right solution.
